src/hammer-vlsi/hammer_vlsi/driver.py

```diff
@@ -806,12 +811,26 @@ def get_hierarchical_settings(self) -> List[Tuple[str, dict]]:
             assert isinstance(list_of_hier_modules, list)
             if len(list_of_hier_modules) == 0:
                 raise ValueError("No hierarchical modules defined manually in manual hierarchical mode")
+            hier_modules = reduce(add_dicts, list_of_hier_modules)
+
             list_of_placement_constraints = self.database.get_setting(
                 "vlsi.inputs.hierarchical.manual_placement_constraints")  # type: List[Dict]
             assert isinstance(list_of_placement_constraints, list)
-            hier_modules = reduce(add_dicts, list_of_hier_modules)
-            combined_raw_placement_dict = reduce(add_dicts, list_of_placement_constraints, {})  # type: Dict[str, dict]
-            hier_placement_constraints = {key: list(map(PlacementConstraint.from_dict, lst))
+            combined_raw_placement_dict = reduce(add_dicts, list_of_placement_constraints, {})  # type: Dict[str, List[Dict[str, Any]]]
+
+            def get_toplevel(d: List[Dict[str, Any]]) -> Optional[Dict[str, Any]]:
```

Comment:
Is this getting all top levels? Not just one? If so maybe pluralize the name.

Reply:
This is just getting one `toplevel` (it gets fed to a map function below).